llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -2641,6 +2641,20 @@ static std::optional<Instruction *> instCombineDMB(InstCombiner &IC,
   return std::nullopt;
 }
 
+static std::optional<Instruction *> instCombinePTrue(InstCombiner &IC,
+                                                     IntrinsicInst &II) {
+  IRBuilder<> Builder(&II);
+  auto Type = cast<VectorType>(II.getType());
+  ConstantInt *Pattern;
+  if (match(II.getOperand(0), m_ConstantInt(Pattern)) &&
+      Pattern->getZExtValue() == AArch64SVEPredPattern::all) {
```

Comment:
```suggestion
  if (match(II.getOperand(0), m_ConstantInt<AArch64SVEPredPattern::all>()) {
```
Not sure but I think this should works.